src/backend/event.ml

```diff
@@ -0,0 +1,117 @@
+        ~description:"Succesful operation"
+        ~content:[
+          "application/json",
+          Spec.make_media_type_object () ~schema:Types.(ref EventIdList.ref);
+        ];
+    ]
+  (* Event query *)
+  |> Router.get "/api/event/:id" get_event
+    ~tags:["event"]
+    ~summary:"Get the details of a single Event"
+    ~parameters:[
+      Types.obj @@ Spec.make_parameter_object ()
+        ~name:"id" ~in_:Path
+        ~description:"Id of the queried event"
+        ~required:true
+        ~schema:Types.(ref EventId.ref)
+    ]
+    ~responses:[
+      "200", Types.obj @@ Spec.make_response_object ()
+        ~description:"Successful operation"
+        ~content:[
+          Spec.json,
+          Spec.make_media_type_object () ~schema:(Types.(ref Event.ref));
+        ];
+      "400", Types.obj @@ Spec.make_error_response_object ()
+        ~description:"Invalid Id supplied";
+      "404", Types.obj @@ Spec.make_error_response_object ()
+        ~description:"Event not found";
+    ]
+  (* Event creation *)
+  |> Router.put "/api/event" create_event
+    ~tags:["event"]
+    ~summary:"Create a new event"
+    ~request_body:(
+      Types.obj @@ Spec.make_request_body_object ()
+        ~description:"Details of the Event to create"
+        ~required:true
+        ~content:[
+          Spec.json,
+          Spec.make_media_type_object () ~schema:(Types.(ref Event.ref));
+        ])
+    ~responses:[
+      "200", Types.obj @@ Spec.make_response_object ()
+        ~description:"Successful operation"
+        ~content:[
+          Spec.json,
+          Spec.make_media_type_object () ~schema:(Types.(ref EventId.ref));
+        ];
+      "400", Types.obj @@ Spec.make_error_response_object ()
+        ~description:"Invalid input";
+    ]
+
+
+(* Event list *)
+(* ************************************************************************* *)
+
+and event_list =
+  Api.get
+    ~to_yojson:Types.EventIdList.to_yojson
+    (fun _req st ->
+      let events = List.map Ftw.Event.id (Ftw.Event.list st) in
+      let res : Types.EventIdList.t = { events; } in
+      Ok res
+    )
+
+(* Event query *)
+(* ************************************************************************* *)
+
+and get_event =
+  Api.get
+    ~to_yojson:Types.Event.to_yojson
+    (fun req st ->
+       let+ id = Utils.int_param req "id" in
+       let+ event =
+         try Ok (Ftw.Event.get st id)
+         with Not_found -> Error.(mk @@ not_found "Event")
+       in
+       let ret : Types.Event.t = {
+         name = Ftw.Event.name event;
+       } in
+       Ok ret
+    )
+
+(* Event creation *)
+(* ************************************************************************* *)
+
+and create_event =
+  Api.put
+    ~of_yojson:Types.Event.of_yojson
+    ~to_yojson:Types.EventId.to_yojson
+    (fun _req st (event : Types.Event.t) ->
+       let start_date = Ftw.Date.mk ~day:1 ~month:1 ~year:2025 in
+       let end_date = Ftw.Date.mk ~day:1 ~month:1 ~year:2025 in
```

Comment:
Do we really want to be able to create an event without providing start and end dates? It seems a bit weird to me

Reply:
I'm adding that right now

Reply:
fixed by 8cd1695